store/storetest/channel_store_categories.go

```diff
@@ -1676,3 +1677,49 @@ func testDeleteSidebarCategory(t *testing.T, ss store.Store, s SqlSupplier) {
 		assert.NotNil(t, err)
 	})
 }
+
+func testUpdateSidebarChannelsByPreferences(t *testing.T, ss store.Store) {
+	t.Run("Should be able to update sidebar channels", func(t *testing.T) {
+		userId := model.NewId()
+		teamId := model.NewId()
+
+		nErr := ss.Channel().CreateInitialSidebarCategories(userId, teamId)
+		require.Nil(t, nErr)
+
+		channel, nErr := ss.Channel().Save(&model.Channel{
+			Name:   "channel",
+			Type:   model.CHANNEL_OPEN,
+			TeamId: teamId,
+		}, 10)
+		require.Nil(t, nErr)
+
+		err := ss.Channel().UpdateSidebarChannelsByPreferences(&model.Preferences{
+			model.Preference{
+				Name:     channel.Id,
+				Category: model.PREFERENCE_CATEGORY_FAVORITE_CHANNEL,
+				Value:    "true",
+			},
+		})
+		assert.NoError(t, err)
+	})
+
+	t.Run("Should return ErrNotFound if channel is not found", func(t *testing.T) {
+		userId := model.NewId()
+		teamId := model.NewId()
+
+		nErr := ss.Channel().CreateInitialSidebarCategories(userId, teamId)
+		assert.Nil(t, nErr)
+
+		err := ss.Channel().UpdateSidebarChannelsByPreferences(&model.Preferences{
+			model.Preference{
+				Name:     "fakeid",
+				Category: model.PREFERENCE_CATEGORY_FAVORITE_CHANNEL,
+				Value:    "true",
+			},
+		})
+		if assert.NotNil(t, err) {
```

Comment:
Should we instead check that a panic hasn't happened like in the other panic tests?

Reply:
Yeah, makes sense.